### PR TITLE
feat(constants): Host Types

### PR DESCRIPTION
**Pull request by souravdasslg**
1. All host types are moved to `host-types.ts`
2. Update all references.

#5088

### CODE REVIEW

**Comment by rarkins** [comment]
`hostType` should be either a datasource or a platform. I think maybe we defer this refactor until after we make those types enumerable.

File: lib/config/env.ts [hunk 1]
```diff
@@ -57,15 +58,15 @@ export function getConfig(env: NodeJS.ProcessEnv): RenovateConfig {
 
   if (env.GITHUB_COM_TOKEN) {
     config.hostRules.push({
-      hostType: 'github',
+      hostType: HOST_TYPE_GITHUB,
```
Comment (rarkins) [inline lib/config/env.ts]:
```suggestion
      hostType: DATASOURCE_GITHUB,
```

File: lib/config/env.ts [hunk 2]
```diff
@@ -57,15 +58,15 @@ export function getConfig(env: NodeJS.ProcessEnv): RenovateConfig {
...
       domainName: 'github.com',
       token: env.GITHUB_COM_TOKEN,
     });
   }
 
   if (env.DOCKER_USERNAME && env.DOCKER_PASSWORD) {
     config.hostRules.push({
-      hostType: 'docker',
+      hostType: HOST_TYPE_DOCKER,
```
Comment (rarkins) [inline lib/config/env.ts]:
```suggestion
      hostType: DATASOURCE_DOCKER,
```

**Review by rarkins — CHANGES_REQUESTED** [review]
Use DATASOURCE_ and PLATFORM_ constants instead

File: lib/config/env.ts (hunk 1 above)
Comment (souravdasslg) [reply lib/config/env.ts]: By default, we will use `DATASOURCE`, in case that is not present, will go for `PLATFORM`

**Comment by souravdasslg** [comment]
Will work on it, after #5179 merged

**Comment by souravdasslg** [comment]
To make things a bit more intuitive, for `azure`, `bitbucket`,`bitbucket-server` `github` and `gitlab` let's use the `platform` constant, and rest of all will be `datasource` or `binarysource`

**Comment by rarkins** [comment]
> To make things a bit more intuitive, for `azure`, `bitbucket`,`bitbucket-server` `github` and `gitlab` let's use the `platform` constant, and rest of all will be `datasource` or `binarysource`

I don't think that `binarySource` is applicable.

File: lib/datasource/metadata.ts [hunk 3]
```diff
@@ -2,6 +2,7 @@ import is from '@sindresorhus/is';
 import parse from 'github-url-from-git';
 import { ReleaseResult } from './common';
 import * as hostRules from '../util/host-rules';
+import { PLATFORM_TYPE_GITHUB } from '../constants/platfroms';
```
Comment (rarkins) [inline lib/datasource/metadata.ts]: I think use DATASOURCE_GITHUB here

File: lib/datasource/pypi/index.ts [hunk 4]
```diff
@@ -1,10 +1,11 @@
 import is from '@sindresorhus/is';
 import url from 'url';
-import { parse } from 'node-html-parser';
+import { HTMLElement, parse } from 'node-html-parser';
```
Comment (rarkins) [inline lib/datasource/pypi/index.ts]:
```suggestion
import { parse } from 'node-html-parser';
```

File: lib/config/env.ts [hunk 5]
```diff
@@ -3,6 +3,8 @@ import is from '@sindresorhus/is';
 import { getOptions, RenovateOptions } from './definitions';
 import { RenovateConfig } from './common';
 import { logger } from '../logger';
+import { DATASOURCE_DOCKER } from '../constants/data-binary-source';
+import { PLATFORM_TYPE_GITHUB } from '../constants/platfroms';
```
Comment (rarkins) [inline lib/config/env.ts]: I just realised that we misspelled "platforms"

**Review by rarkins — CHANGES_REQUESTED** [review]

**Comment by rarkins** [comment]
@souravdasslg could you do the platfroms/platforms rename in a separate PR? Will be easier to merge that first.

**Review by rarkins — CHANGES_REQUESTED** [review]
Separate platfroms rename

File: lib/datasource/github/index.ts [hunk 6]
```diff
@@ -8,6 +8,7 @@ import {
 import { logger } from '../../logger';
 import got, { GotJSONOptions } from '../../util/got';
 import { PLATFORM_FAILURE } from '../../constants/error-messages';
+import { PLATFORM_TYPE_GITHUB } from '../../constants/platforms';
```
Comment (rarkins) [inline lib/datasource/github/index.ts]:
```suggestion
import { DATASOURCE_GITHUB } from '../../constants/data-binary-source';
```

File: lib/datasource/github/index.ts [hunk 7]
```diff
@@ -20,7 +21,7 @@ async function fetchJSONFile(repo: string, fileName: string): Promise<Preset> {
         : 'application/vnd.github.v3+json',
     },
     json: true,
-    hostType: 'github',
+    hostType: PLATFORM_TYPE_GITHUB,
```
Comment (rarkins) [inline lib/datasource/github/index.ts]:
```suggestion
    hostType: DATASOURCE_GITHUB,
```

**Review by rarkins — CHANGES_REQUESTED** [review]

**Pull request merged**